lib/shopify_api/resources/base.rb

```diff
@@ -63,12 +66,61 @@ def clear_session
         self.headers.delete('X-Shopify-Access-Token')
       end
 
+      def api_prefix
+        API_PREFIX
+      end
+
+      def prefix(options = {})
+        "#{api_prefix}#{resource_prefix(options)}"
+      end
+
+      def prefix_source
+        ''
+      end
+
+      def resource_prefix(_options = {})
+        ''
+      end
+
+      # Sets the \prefix for a resource's nested URL (e.g., <tt>prefix/collectionname/1.json</tt>).
+      # Default value is <tt>site.path</tt>.
+      def resource_prefix=(value)
```

Comment:
does this method need additional testing or is it covered by existing ones?

Reply:
it is covered by existing tests,  all the urls would be broken with out it.